cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MessagingMetricsProcessorService.java

```diff
@@ -245,6 +288,14 @@ protected void run() {
         Thread.currentThread().interrupt();
       }
     }
+
+    try {
```

Comment:
maybe safer not to join, but set it to daemon and just exit. Otherwise the join my make the process hang on shutdown